.github/workflows/build.yml

```diff
@@ -0,0 +1,48 @@
+name: Build
+on:
+  pull_request:
+  push:
+    branches:
+      - "main"
+    tags:
+      - "v*"
+  release:
+    types:
+      - published
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```

Comment:
you need

```yml
with:
  ref: ${{ github.ref }}
  fetch-depth: 0
```

everywhere because of setuptools-scm

Reply:
Why do we need this anywhere else apart from in the `build.yml` workflow? `setuptools-scm` will fallback to `0.0.0`, right?

Reply:
i set both options for all checkout